var/spack/repos/builtin/packages/hiop/package.py

```diff
@@ -64,10 +65,16 @@ class Hiop(CMakePackage, CudaPackage, ROCmPackage):
 
     depends_on('magma+cuda', when='+cuda')
     depends_on('magma+rocm', when='+rocm')
-    depends_on('magma@2.5.4:', when='@0.4:+cuda')
-    depends_on('magma@2.6.1:', when='@0.4.6:+cuda')
-    depends_on('magma@2.5.4:', when='@0.4:+rocm')
-    depends_on('magma@2.6.1:', when='@0.4.6:+rocm')
+
+    # Depends on Magma when +rocm or +cuda
+    magma_ver_constraints = (
+        ('2.5.4', '0.4'),
+        ('2.6.1', '0.4.6'),
+        ('2.6.2', '0.5.4'),
+    )
+    for (magma_v, hiop_v) in magma_ver_constraints:
+        depends_on('magma@{0}:'.format(magma_v), when='@{0}:+cuda'.format(hiop_v))
+        depends_on('magma@{0}:'.format(magma_v), when='@{0}:+rocm'.format(hiop_v))
```

Comment:
I'm conflicted about this. This is very clean, but I wish there was syntax along the lines of the following:

```python
depends_on('magma@version:', when='^magma')
```

Realistically this is a small gripe, but it is frustrating having to re-specify dependency logic when attempting to define version constraints.

Reply:
See #14002